operator/src/main/java/org/keycloak/operator/OperatorManagedResource.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.operator;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.OwnerReference;
+import io.fabric8.kubernetes.api.model.OwnerReferenceBuilder;
+import io.fabric8.kubernetes.client.CustomResource;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.quarkus.logging.Log;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Represents a single K8s resource that is managed by this operator (e.g. Deployment, Service, Ingress, etc.)
+ *
+ * @author Vaclav Muzikar <vmuzikar@redhat.com>
+ */
+public abstract class OperatorManagedResource {
```

Comment:
It is overlapping for sure, but since "Dependent Resources" is now still just a [PR](https://github.com/java-operator-sdk/java-operator-sdk/pull/785) and not something we can use right now, I decided to use an interim solution. We can (and we should!) refactor later.

Reply:
deal 👍

Reply:
can you mark this as resolved @vmuzikar